src/base-commerce-command.js

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2021 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+const { Command } = require('@oclif/command')
+const { cli } = require('cli-ux')
+const { initSdk } = require('./cloudmanager-helpers')
+class BaseCommerceCommand extends Command {
+  async runSync (programId, environmentId, body, pollingInterval, imsContextName = null) {
+    const sdk = await initSdk(imsContextName)
+    const postResponse = await sdk.postCLICommand(programId, environmentId, body)
+
+    const { id: commandId } = await postResponse
+
+    let result = await this.callGet(sdk, programId, environmentId, commandId)
+
+    while (result.status === 'RUNNING' || result.status === 'CREATING') {
+      await cli.wait(pollingInterval)
+      result = await this.callGet(sdk, programId, environmentId, commandId, true)
```

Comment:
Same comment as above -- presumably this can throw an error. Although in this context, it's a bit stranger since the action wouldn't be stopped. I'm not sure what happens in that case.

Reply:
Yes. Will be caught in subclass.